sparse/coo/common.py

```diff
@@ -244,44 +240,38 @@ def kron(a, b):
            [0, 0, 0, 1, 2, 3, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 1, 2, 3]], dtype=int64)
     """
-    from .core import COO, as_coo
-    check_zero_fill_value(a, b)
+    from .core import COO
+    from .umath import _cartesian_product
 
-    a_sparse = isinstance(a, SparseArray)
-    b_sparse = isinstance(b, SparseArray)
+    a_sparse = isinstance(a, (SparseArray, scipy.sparse.spmatrix))
+    b_sparse = isinstance(b, (SparseArray, scipy.sparse.spmatrix))
```

Comment:
We don't ever test with a `scipy.sparse.spmatrix`, is this something that should also be added to the tests?

Reply:
Added a test.

Reply:
For future references, `asCOO` and the `COO` constructor are pretty heavily tested, so we probably don't need to test things like these everywhere.